graphql/src/values/mod.rs

```diff
@@ -1,15 +1,77 @@
-use graphql_parser::query::Value;
+use graphql_parser::query::{Number, Value};
 use std::collections::BTreeMap;
 use std::iter::FromIterator;
 
-/// Utilties for coercing GraphQL values based on GraphQL types.
+/// Utilities for coercing GraphQL values based on GraphQL types.
 pub mod coercion;
 
 pub use self::coercion::MaybeCoercible;
 
 /// Creates a `graphql_parser::query::Value::Object` from key/value pairs.
+/// If you don't need to determine which keys are included dynamically at runtime
+/// consider using the `object! {}` macro instead.
 pub fn object_value(data: Vec<(&str, Value)>) -> Value {
     Value::Object(BTreeMap::from_iter(
         data.into_iter().map(|(k, v)| (k.to_string(), v)),
     ))
 }
+
+pub trait IntoValue {
+    fn into_value(self) -> Value;
+}
+
+impl IntoValue for Value {
+    #[inline]
+    fn into_value(self) -> Value {
+        self
+    }
+}
+
+impl<T: IntoValue> IntoValue for Option<T> {
+    #[inline]
+    fn into_value(self) -> Value {
+        match self {
+            Some(v) => v.into_value(),
+            None => Value::Null,
+        }
+    }
+}
+
+macro_rules! impl_into_values {
+    ($(($T:ty, $V:ident)),*) => {
+        $(
+            impl IntoValue for $T {
+                #[inline]
+                fn into_value(self) -> Value {
+                    Value::$V(self)
+                }
+            }
+        )+
+    };
+}
+
+impl_into_values![
+    (String, String),
+    (f64, Float),
+    (bool, Boolean),
+    (Vec<Value>, List),
+    (Number, Int)
+];
+
+/// Creates a `graphql_parser::query::Value::Object` from key/value pairs.
+#[macro_export]
+macro_rules! object {
```

Comment:
Neat! That's a nice tool to tighten up some of our code. At least for our tests, would also be nice to have an `entity!` macro along the same lines.

Reply:
I'd be happy to create one. Can you point me to one test that would use this macro so I know what you're looking for?

Reply:
Nevermind, I think you mean just the search for `Entity::new`

Reply:
It's actually pretty easy since `graph::data::store::Value` already implements `From` for a lot of cases:
```rust
macro_rules! entity {
    ($($name:ident: $value:expr,)*) => {
        {
            let mut result = ::graph::prelude::Entity::new();
            $(
                result.insert(stringify!($name).to_string(), Value::from($value));
            )*
            result
        }
    };
    ($($name:ident: $value:expr),*) => {
        entity! {$($name: $value,)*}
    };
}
```